db/snapshot.go

```diff
@@ -18,9 +18,18 @@ func (c *Collection) GetSnapshot() (*Snapshot, error) {
 	if err != nil {
 		return nil, err
 	}
+	c.indexMut.RLock()
+	indexes := make([]*Index, len(c.indexes))
+	copy(indexes, c.indexes)
+	defer c.indexMut.RUnlock()
 	return &Snapshot{
-		readOnlyCollection: c.readOnlyCollection,
-		snapshot:           snapshot,
+		readOnlyCollection: &readOnlyCollection{
```

Comment:
The main cause of the bug was that I forgot to create a new `readOnlyCollection` here with the `snapshot` as the reader.